cumulus_library/statistics/psm.py

```diff
@@ -0,0 +1,247 @@
+        )
+        super().__init__()
+
+    def _get_symptoms_dict(self, path: str) -> dict:
+        """convenience function for loading symptoms dictionaries from a json file"""
+        with open(path) as f:
+            symptoms = json.load(f)
+        return symptoms
+
+    def _get_sampled_ids(
+        self,
+        cursor: DatabaseCursor,
+        schema: str,
+        query: str,
+        sample_size: int,
+        dependent_variable: str,
+        is_positive: bool,
+    ):
+        """Creates a table containing randomly sampled patients for PSM analysis
+
+        To use this, it is assumed you have already identified a cohort of positively
+        IDed patients as a manual process.
+        :param cursor: A valid DatabaseCusror:
+        :param schema: the schema/database name where the data exists
+        :param query: a query generated from the psm_dsitinct_ids template
+        :param sample_size: the number of records to include in the random sample.
+            This should generally be >= 20.
+        :param dependent_variable: the name to use for your filtering column
+        :param is_positive: defines the value to be used for your filtering column
+        """
+        df = cursor.execute(query).as_pandas()
+        df = (
+            df.sort_values(by=[self.config.primary_ref])
+            .reset_index()
+            .drop("index", axis=1)
+        )
+        df = (
+            # TODO: remove replace behavior after increasing data sample size
+            df.sample(n=sample_size, random_state=self.config.seed, replace=True)
+            .sort_values(by=[self.config.primary_ref])
+            .reset_index()
+            .drop("index", axis=1)
+        )
+
+        df[dependent_variable] = is_positive
+        return df
+
+    def _create_covariate_table(self, cursor: DatabaseCursor, schema: str):
+        """Creates a covariate table from the loaded toml config"""
+        # checks for primary & link ref being the same
+        source_refs = list(
+            {self.config.primary_ref, self.config.count_ref} - set([None])
+        )
+        pos_query = get_distinct_ids(source_refs, self.config.pos_source_table)
+        pos = self._get_sampled_ids(
+            cursor,
+            schema,
+            pos_query,
+            self.config.pos_sample_size,
+            self.config.dependent_variable,
+            1,
+        )
+        neg_query = get_distinct_ids(
+            source_refs,
+            self.config.neg_source_table,
+            join_id=self.config.primary_ref,
+            filter_table=self.config.pos_source_table,
+        )
+        neg = self._get_sampled_ids(
+            cursor,
+            schema,
+            neg_query,
+            self.config.neg_sample_size,
+            self.config.dependent_variable,
+            0,
+        )
+        cohort = pandas.concat([pos, neg])
+
+        # Replace table (if it exists)
+        # TODO - replace with timestamp prepended table
```

Comment:
Just pointing this todo out to confirm it's an intentional post-PR comment, not an internal-PR comment.